### PR TITLE
fix: handle null as the first argument in identify

### PR DESCRIPTION
**Pull request by oscb**
- [x] I've included a changeset (psst. run `yarn changeset`. Read about changesets [here](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md)).

### CODE REVIEW

File: packages/browser/src/core/arguments-resolver/index.ts [hunk 1]
```diff
@@ -111,22 +111,56 @@ export const resolveUserArguments = <T extends Traits, U extends User>(
...
-    const objects = args.filter((obj) => {
-      if (id === null) {
-        return isPlainObject(obj)
+    const values: {
+      id?: ID
+      traits?: T | null
+      options?: Options
+      callback?: Callback
+    } = {}
+    // It's a stack so it's reversed so that we go through each of the expected arguments
```
Comment (silesky) [inline packages/browser/src/core/arguments-resolver/index.ts:120]: Is this a leetcode problem 😁

Comment (oscb) [reply packages/browser/src/core/arguments-resolver/index.ts:120]: It totally is

**Review by chrisradek — APPROVED** [review]

File: packages/browser/src/core/arguments-resolver/index.ts (hunk 1 above)
Comment (bsneed) [reply packages/browser/src/core/arguments-resolver/index.ts:120]: I f'ing love this.

File: packages/browser/src/core/arguments-resolver/__tests__/index.test.ts [hunk 2]
```diff
@@ -429,6 +429,33 @@ describe(resolveUserArguments, () => {
...
+    expect(traits).toEqual(userTraits)
+    expect(options).toEqual({})
+    expect(id).toEqual(null)
+  })
+
+  it('should accept (null, traits) when identity is set', () => {
+    user.reset()
+    user.identify('something')
+    const [id, traits, options] = resolver(null, userTraits)
+    console.log(id, traits, options)
```
Comment (silesky) [inline packages/browser/src/core/arguments-resolver/__tests__/index.test.ts:454]: console.log

File: packages/browser/src/core/arguments-resolver/__tests__/index.test.ts [hunk 3]
```diff
@@ -429,6 +429,33 @@ describe(resolveUserArguments, () => {
...
+    const [id, traits, options] = resolver(undefined, userTraits)
+    expect(traits).toEqual(userTraits)
+    expect(options).toEqual({})
+    expect(id).toEqual(null)
+  })
+
+  it('should accept (null, traits) with unknown identity', () => {
+    user.reset()
+    const [id, traits, options] = resolver(null, userTraits)
+    console.log(id, traits, options)
```
Comment (silesky) [inline packages/browser/src/core/arguments-resolver/__tests__/index.test.ts:444]: console.log

File: packages/browser/src/core/arguments-resolver/__tests__/index.test.ts (hunk 2 above)
Comment (oscb) [reply packages/browser/src/core/arguments-resolver/__tests__/index.test.ts:454]: oops!

File: packages/browser/src/core/arguments-resolver/index.ts [hunk 4]
```diff
@@ -111,22 +111,56 @@ export const resolveUserArguments = <T extends Traits, U extends User>(
...
+    const orderStack: Array<keyof typeof values> = [
```
Comment (silesky) [inline packages/browser/src/core/arguments-resolver/index.ts:121]: Using the old TS array syntax???

**Review by silesky — APPROVED** [review]
LGTM

**Pull request merged**